sailor/assetcentral/functional_location.py

```diff
@@ -0,0 +1,123 @@
+    _AssetcentralField('_changed_on', 'changedOn', get_extractor=_string_to_timestamp_parser(unit='ms')),
+    _AssetcentralField('_published_on', 'publishedOn', get_extractor=_string_to_timestamp_parser(unit='ms')),
+    _AssetcentralField('_tag_number', 'tagNumber'),
+    _AssetcentralField('_lifecycle', 'lifeCycle'),
+    _AssetcentralField('_lifecycle_description', 'lifeCycleDescription'),
+    _AssetcentralField('_source', 'source'),
+    _AssetcentralField('_image_URL', 'imageURL'),
+    _AssetcentralField('_coordinates', 'coordinates'),
+    _AssetcentralField('_floc_status', 'flocStatus'),
+    _AssetcentralField('_is_operator_valid', 'isOperatorValid'),
+    _AssetcentralField('_model_version', 'modelVersion'),
+    _AssetcentralField('_sold_to', 'soldTo'),
+    _AssetcentralField('_image', 'image'),
+    _AssetcentralField('_consume', 'consume'),
+    _AssetcentralField('_dealer', 'dealer'),
+    _AssetcentralField('_service_provider', 'serviceProvider'),
+    _AssetcentralField('_primary_external_id', 'primaryExternalId'),
+    _AssetcentralField('_floc_search_terms', 'flocSearchTerms'),
+    _AssetcentralField('_source_search_terms', 'sourceSearchTerms'),
+    _AssetcentralField('_manufacturer_search_terms', 'manufacturerSearchTerms'),
+    _AssetcentralField('_operator_search_terms', 'operatorSearchTerms'),
+    _AssetcentralField('_class', 'class')
+]
+
+
+@_base.add_properties
+class FunctionalLocation(AssetcentralEntity):
+    """AssetCentral Functional Location Object."""
+
+    _field_map = {field.our_name: field for field in _FUNCTIONAL_LOCATION_FIELDS}
+
+
+class FunctionalLocationSet(AssetcentralEntitySet):
+    """Class representing a group of Functional Locations."""
+
+    _element_type = FunctionalLocation
+    _method_defaults = {
+        'plot_distribution': {
+            'by': 'model_name',
+        },
+    }
+
+
+def find_functional_locations(*, extended_filters=(), **kwargs) -> FunctionalLocationSet:
+    """
+    Fetch Functional Locations from AssetCentral with the applied filters, return an FunctionalLocationSet.
+
+    This method supports the common filter language explained at :ref:`filter`.
+
+    Parameters
+    ----------
+    extended_filters
+        See :ref:`filter`.
+    **kwargs
+        See :ref:`filter`.
+
+    Examples
+    --------
+    Find all Functional Locations with the name 'MyFloc'::
+
+        find_functional_locations(name='MyFloc')
+
+    Find all Functional Locations which either have the name 'MyFloc' or the name 'MyOtherFloc'::
+
+        find_functional_locations(name=['MyFloc', 'MyOtherFloc'])
+
+    Find all Functional Locations with the name 'MyFloc' which are also located in 'London'::
+
+        find_functional_locations(name='MyFloc', location_name='London')
+
+    Find all Functional Locations created between January 1, 2018 and January 1, 2019 in 'London'::
+
+        find_functional_locations(extended_filters=['_created_on >= "2018-01-01"', '_created_on < "2019-01-01"'],
```

Comment:
I'd prefer not to have any 'private' fields (fields starting with an `_` are not part of the stable API) in these documentation examples.
Two options:
- just remove this last example, I don't think it's necessary
- If you have a use-case that actually requires `_created_on` rename the field in the field map to `created_on`